src/psyclone/psyir/transformations/intrinsics/sum2code_trans.py

```diff
@@ -144,264 +157,68 @@ class Sum2CodeTrans(Transformation):
     <BLANKLINE>
 
     '''
-    @staticmethod
-    def _get_args(node):
-        '''Utility method that returns the sum arguments, (array reference,
-        dimension and mask).
-
-        :param node: a Sum intrinsic.
-        :type node: :py:class:`psyclone.psyir.nodes.IntrinsicCall`
-
-        returns: a tuple containing the 3 sum arguments.
-        rtype: Tuple[py:class:`psyclone.psyir.nodes.reference.Reference`, \
-                     py:class:`psyclone.psyir.nodes.Literal` | \
-                     :py:class:`psyclone.psyir.nodes.Reference`, \
-                     Optional[:py:class:`psyclone.psyir.nodes.node`]]
+    _INTRINSIC_NAME = "SUM"
+
+    def _loop_body(self, array_reduction, array_iterators, var_symbol,
+                   array_ref):
+        '''Provide the body of the nested loop that computes the sum of an
+        array.
+
+        :param bool array_reduction: True if the implementation should
+            provide a sum over a particular array dimension and False
+            if the sum is for all elements of the array.
+        :param array_iterators: a list of datasymbols containing the
+            loop iterators ordered from outermost loop symbol to innermost
+            loop symbol.
+        :type array_iterators:
+            List[:py:class:`psyclone.psyir.symbols.DataSymbol`]
+        :param var_symbol: the symbol used to store the final result.
+        :type var_symbol: :py:class:`psyclone.psyir.symbols.DataSymbol`
+        :param array_ref: a reference to the array from which the
```

Comment:
s/from which/for which/

Reply:
Modified here and in the other two transformations.